src/brsTypes/components/RoSGNode.ts

```diff
@@ -0,0 +1,222 @@
+        if (index.kind !== ValueKind.String) {
+            throw new Error("RoSGNode indexes must be strings");
+        }
+        this.elements.set(index.value.toLowerCase(), value);
+        return BrsInvalid.Instance;
+    }
+
+    /** Removes all elements from the node */
+    private clear = new Callable("clear", {
+        signature: {
+            args: [],
+            returns: ValueKind.Void,
+        },
+        impl: (interpreter: Interpreter) => {
+            this.elements.clear();
+            return BrsInvalid.Instance;
+        },
+    });
+
+    /** Removes a given item from the node */
+    private delete = new Callable("delete", {
+        signature: {
+            args: [new StdlibArgument("str", ValueKind.String)],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter, str: BrsString) => {
+            let deleted = this.elements.delete(str.value);
+            return BrsBoolean.from(deleted);
+        },
+    });
+
+    /** Given a key and value, adds an item to the node if it doesn't exist
+     * Or replaces the value of a key that already exists in the node
+     */
+    private addreplace = new Callable("addreplace", {
+        signature: {
+            args: [
+                new StdlibArgument("key", ValueKind.String),
+                new StdlibArgument("value", ValueKind.Dynamic),
+            ],
+            returns: ValueKind.Void,
+        },
+        impl: (interpreter: Interpreter, key: BrsString, value: BrsType) => {
+            this.set(key, value);
+            return BrsInvalid.Instance;
+        },
+    });
+
+    /** Returns the number of items in the node */
+    private count = new Callable("count", {
+        signature: {
+            args: [],
+            returns: ValueKind.Int32,
+        },
+        impl: (interpreter: Interpreter) => {
+            return new Int32(this.elements.size);
+        },
+    });
+
+    /** Returns a boolean indicating whether or not a given key exists in the node */
+    private doesexist = new Callable("doesexist", {
+        signature: {
+            args: [new StdlibArgument("str", ValueKind.String)],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter, str: BrsString) => {
+            return this.get(str) !== BrsInvalid.Instance ? BrsBoolean.True : BrsBoolean.False;
+        },
+    });
+
+    /** Appends a new node to another. If two keys are the same, the value of the original AA is replaced with the new one. */
+    private append = new Callable("append", {
+        signature: {
+            args: [new StdlibArgument("obj", ValueKind.Object)],
+            returns: ValueKind.Void,
+        },
+        impl: (interpreter: Interpreter, obj: BrsType) => {
+            if (obj instanceof RoAssociativeArray || obj instanceof RoSGNode) {
+                this.elements = new Map<string, BrsType>([
+                    ...this.elements,
+                    ...(obj as any).elements,
```

Comment:
Similar to the above, I added this to avoid a type error, but I'm less sure what the problem is here.  Even when just checking for an RoSGNode instance I still get this error.

Reply:
This one was caused by the fact that `elements` wasn't public on `RoAssociativeArray`.  It is now!